cpp/src/io/orc/reader_impl.hpp

```diff
@@ -71,18 +66,50 @@ class reader::impl {
                            std::vector<std::vector<size_type>> const& stripes);
 
  private:
+  /**
+   * @brief Perform the necessary data preprocessing for parsing file later on.
+   *
+   * @param skip_rows Number of rows to skip from the start
+   * @param num_rows_opt Optional number of rows to read, or `std::nullopt` to read all rows
+   * @param stripes Indices of individual stripes to load if non-empty
+   */
+  void prepare_data(uint64_t skip_rows,
+                    std::optional<size_type> const& num_rows_opt,
+                    std::vector<std::vector<size_type>> const& stripes);
+
+  /**
+   * @brief Populate the output table metadata from the file metadata.
+   *
+   * @return Columns' metadata to output along with the table read from file
+   */
+  table_metadata populate_metadata();
```

Comment:
The name doesn't quite fit the singature. `make_table_metadata` would be a batter option IMO.

Reply:
Tried to match it with the similar function in parquet reader. How about `make_output_metadata`?

Reply:
Anything that makes sense with factory-ness of the function is fine by me